flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -217,6 +217,8 @@
 
 	private Long syncSavepointId = null;
 
+	private long latestAsyncCheckpointStartDelayNanos;
```

Comment:
For the full correctness yes, but:

1. we long decided that we don't want to sacrifice performance for metrics (there are other examples of that - here it's not that important, but take a look for example at `RecordWriter#numBytesOut` or `RecordWriter#numBuffersOut`, which are on more performance sensitive path)
2. metrics reading is not critical, even if gets older/incorrect value
3. I haven't seen problems caused by this so far
4. the proper solution that wouldn't impact performance as much would be to read the metrics from the mailbox thread (via mailbox action). It would be a zero overhead on the metric collecting path, small and very rare overhead for the task thread during measurement and some extra overhead for the thread asking for metric values (but that's not important)